internal/charmstore/store.go

```diff
@@ -215,7 +222,10 @@ func (p *Pool) requestStoreNB(always bool) (*Store, error) {
 		bp.Store = macStore
 		bsvc, err := bakery.NewService(bp)
 		if err != nil {
-			return nil, errgo.Notef(err, "cannot make bakery service")
+			// This should never happen because the only reason bakery.NewService
```

Comment:
One way of fixing that (backwardly incompatibly, unfortunately) is by changing bakery.NewService so it doesn't return an error, and provide a function to fill out bakery parameter defaults. There's really no reason why it should return an error, even in the future, as it doesn't interact with anything external. I may have to eat my words at some point in the future though :)